src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -470,7 +470,7 @@ trait TypeAssigner {
     tree.withType(sym.nonMemberTermRef)
 
   def assignType(tree: untpd.Annotated, annot: Tree, arg: Tree)(implicit ctx: Context) =
-    tree.withType(AnnotatedType(arg.tpe, Annotation(annot)))
+    tree.withType(AnnotatedType(arg.tpe.widen, Annotation(annot)))
```

Comment:
I have a hunch that this change will disable things like `42.type` (SIP 23). If Dotty supports it in the future, we'll need to revisit this change...